### ISSUE TITLE
When shoryuken cannot find queues, dont spam the log, just crash.

### ISSUE THREAD

**Issue by Senjai** [phstc/shoryuken#172]
To our fault, shoryuken workers were started on a box without the permissions required to access the SQS queues we have setup for them. This was remedied, but when shoryuken experiences this issue messages like this:

```
2016-01-09T16:31:59Z 77103 TID-1tn6x8 ERROR: Error fetching message: The specified queue does not exist or you do not have access to it.
```

Begin spamming the log. So much so that I removed a 60gb log file this morning. This is not considered exceptional, nor does it crash the application (one of which should be true). For companies with log quotas, this can be financially problematic. 

Could we crash if we get these issues? It is a configuration issue and should halt the application before working on queues.

**Comment by zbintliff** [comment]
Imagine have 4 different consumers running and only 1 can't reach queues.  There is no reason to crash and stop other 3 queues from trying to work.

If the log file is the problem there are system tools like logrotate and tidy that can help you.

**Comment by Senjai** [comment]
@zbintliff I think you missed the point. We use logrotate, this can generate a log of that volume before the logs are scheduled to be rotated. Rotating logs also doesn't remove them from disk, it still takes space. Also if your using logentries, or a splunk drain, exceeding the amount of bandwidth of logs sent to them is expensive.

I understand your logic. But all 4 queues are known at runtime, and each can be validated with a get_queue_url request before starting workers at startup. I would even be fine with the ability for a queue to crash once, report a message/trigger an exception and let the others work. 

But saying that the equivalent of:
```ruby
loop do
  log "Looping for the lols"
end
``` 
is an okay outcome when this behaviour occurs is pretty silly.

**Comment by jemmyw** [comment]
+1, I'd like the process to crash if it can't find all of it's queues. If that happens then something has gone wrong, and monitoring software will pick up that the worker isn't running.

Perhaps we could have a lifecycle event so we could say ```on(:error) { exit(1) }```

**Comment by phstc** [comment]
Hi all,

I've just merged in a fix for that #196 - @laCour great work! 🍻 

I'm wondering if it deserves a major release, to avoid breaking changes for people relying on that warning. But I'm not sure if it's a real use case. WDYT?
